bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java

```diff
@@ -1183,6 +1183,40 @@ void setResizeChildren (boolean resize) {
 	}
 }
 
+@Override
+public void setRedraw(boolean redraw) {
+	/*
+	 * Some embedded applications like webview2 (Edge), Excel, etc have a renderer
+	 * that will draw the whole application white when setting redraw to false and
+	 * will repaint it completely when setting it to true. This causes flickering,
+	 * which is why turning redraw off is not allowed in case this control has an
+	 * embedded application as a child.
+	 *
+	 * https://github.com/eclipse-platform/eclipse.platform.swt/issues/1122
+	 */
+	boolean isEmbedded = (style & SWT.EMBEDDED) != 0;
+	if (!redraw && (isEmbedded || hasEmbeddedChildren())) {
+		return;
+	}
+	super.setRedraw(redraw);
+}
+
+private boolean hasEmbeddedChildren() {
+	for (Control child : getChildren()) {
+		if (!child.isDisposed() && child.isVisible()) {
+
+			if ((child.style & SWT.EMBEDDED) != 0)
+				return true;
+
+			// if it's a composite, check its children too
+			if (child instanceof Composite comp && comp.hasEmbeddedChildren())
+				return true;
```

Comment:
Can we have the expression blocks encapsulated in braces?

Reply:
The code has been removed.